dask/bag/tests/test_bag.py

```diff
@@ -1605,3 +1612,38 @@ def test_dask_layers_to_delayed(optimize):
 
     with pytest.raises(ValueError, match="not in"):
         db.Item(arr.dask, (arr.name,), layer="foo")
+
+
+def test_to_dataframe_optimize_graph():
+    pytest.importorskip("dask.dataframe")
+    from dask.dataframe.utils import assert_eq as assert_eq_df
+
+    x = db.from_sequence(
+        [{"name": "test1", "v1": 1}, {"name": "test2", "v1": 2}], npartitions=2
+    )
+
+    # linear operations will be fused by graph optimization
+    with dask.annotate(foo=True):
+        y = x.map(lambda a: dict(**a, v2=a["v1"] + 1))
+        y = y.map(lambda a: dict(**a, v3=a["v2"] + 1))
+
+    # verifying the maps are not fused yet
+    assert len(y.dask) > y.npartitions
```

Comment:
```suggestion
    assert len(y.dask) == y.npartitions * 3
```
Since what you had before is actually true whether or not they were fused